modules/aws-eks-managed-node-groups/templates/userdata-amazonlinux2eks.tpl

```diff
@@ -11,7 +11,8 @@ echo "Running custom user data script"
 # User-supplied pre userdata
 ${pre_userdata}
 
-# Format and Mount NVMe Disks if available
+if [ ${format_mount_nvme_disk} = true ];then
```

Comment:
Question: should we have this baked into the module or should we just demonstrate how users can add this to `pre_userdata` through an example? Any changes here will be impacting to both users who utilize this functionality or not (until [this](https://github.com/hashicorp/terraform-provider-aws/pull/23604) is supported by launch templates)

Reply:
I thought about moving this to pre-userdata but we already have this block of code for both managed and self-managed launch templates. So, it's easier to add a condition with a new variable.

This change wont impact the existing node groups unless they upgrade their node group which creates a new LT with this change. We can make this  to `true` by default so that it wont impact even if the upgrade happens. ONDAT team may need to send `format_mount_nvme_disk=false` for their usecase.

Reply:
Right, the impact of formatting or not formatting won't have an impact, but I'm talking more generically in terms of any changes (ANY changes) within user data means the hash is changed and there will be a new diff that launch templates will want to roll out. I haven't looked at what our default strategy is for launch template updates, but this is one of the reasons I fought very hard to keep any customziations out of the provided user data because it affects everyone (a whitespace change will affect everyone) and the more customizations there are, the more updates to the user data, and the more changes felt by users. Just something to consider if this is not expected to be used by 80%+ of blueprints users but only a select few